docs/book/src/developer/architecture/controllers/cluster.md

```diff
@@ -11,6 +11,17 @@ The Cluster controller's main responsibilities are:
 
 ## Contracts
 
+### Relationship to other Cluster API types
```

Comment:
What are you trying to surface, and this is super useful, but I'm wondering if this is there right place to do it given that here we are talking about the Cluster controller, while those informations most probably are important when trying to understand the InfrastructureCluster controller (down in this page) and the control plane controller (which is documented on another page).

What about adding a "Relationship with other controllers" paragraph under those, so we split control plane and infrastructure?

Reply:
I was trying to have an analogous to https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/architecture/controllers/control-plane.md#relationship-to-other-cluster-api-types.
The way the Cluster CR coordinates with infraCluster and controlPlane and its dependencies are not particularly intuitive for people with little context, so I thought it'd be useful to have them clarified in one single place.

>What about adding a "Relationship with other controllers" paragraph under those, so we split control plane and infrastructure?

you mean having two sections one here another one under control-plane.md? sgtm though I think having this to be read as above in a single place somewhere it'd be illustrative for new joiners to understand how all pieces play together.

Reply:
Ok, I understand the value of providing an overall documentation of Cluster controller interactions, but my concern is that the main consumers of this documentation are providers developers, and we are not covering their perspective given they start from control plane or infrastructure cluster docs.

What about:
- in this paragraph (or better immediately under Contracts without a sub title) just state that the Cluster controller interacts with InfrastructureCluster and the control plane controller, linking the corresponding paragraphs in the the two controllers for details.
- in the corresponding paragraphs for InfrastructureCluster and the control plane controller, provide details of the interactions each controller has with the Cluster/Controller; this will allow to extend those details e.g. by adding interaction between control plane and MHC etc. etc. thus providing a single point where the providers developers can find relevant info for each controller type

Reply:
Adding a little bit more context, also CABPK and probably other controllers interact with the Cluster for ownership/infrastructure ready etc. etc. and I'm a little bit of conflating all those infos in a single paragraph; this is another reason for moving details where their are consumed.

Reply:
ok, updated to complement both Infrastructure Provider and Control Plane Provider sections.
PTAL.